python/pylibcudf/pylibcudf/aggregation.pyx

```diff
@@ -122,6 +122,108 @@ cdef class Aggregation:
     def __hash__(self):
         return dereference(self.c_obj).do_hash()
 
+    def __reduce__(self):
+        cdef std_var_aggregation *std_var_cast
+        cdef quantile_aggregation *quantile_cast
+        cdef nunique_aggregation *nunique_cast
+        cdef nth_element_aggregation *nth_element_cast
+        cdef ewma_aggregation *ewma_cast
+        cdef rank_aggregation *rank_cast
+        cdef collect_list_aggregation *collect_list_cast
+        cdef collect_set_aggregation *collect_set_cast
+        cdef udf_aggregation *udf_cast
+        cdef correlation_aggregation *correlation_cast
+        cdef covariance_aggregation *covariance_cast
+
+        if self.kind() is Kind.SUM:
```

Comment:
Can we replace this cascade with a singledispatch helper function? It's a lot of clauses to fail through in the worst case.